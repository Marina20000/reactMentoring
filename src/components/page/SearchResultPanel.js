import React from 'react';
import style from './style.css';
import Inscription from './../captions/Inscription';
import CreateButton from './../buttons/CreateButton';


export default class SearchResultPanel extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        let quantity = !!this.props.quantity ? this.props.quantity : 0;
        let showPage2 = !!this.props.showPage2 ? this.props.showPage2 : false;
        return (
            <div className={style.searchResultPanel}>
                {!showPage2 && <Inscription className={style.quantityMovies} inscription={`${quantity} movies found`} />}
                {!showPage2 && <CreateButton onClick={() => { this.props.toggleSortings() }} className={this.props.raiting ? style.ratingButtonSelected : style.ratingButtonNoneSelected } inscription='raiting' />}
                {!showPage2 && <CreateButton onClick={() => { this.props.toggleSortings() }} className={this.props.releaseDate ? style.reliaseDateButtonNoneSelected : style.reliaseDateButtonSelected} inscription='reliase date' />}
                {!showPage2 && <Inscription className={style.sortBy} inscription='Sort by' />}
            </div>
        )
    }
}
