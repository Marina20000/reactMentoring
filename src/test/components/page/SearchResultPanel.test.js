import React from "react";
import renderer from 'react-test-renderer';
import SearchResultPanel from './../../../components/page/SearchResultPanel';

describe('SearchResultPanel', () => {
    it("testing with snapshot", () => {
        const tree = renderer.create(<SearchResultPanel/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})

