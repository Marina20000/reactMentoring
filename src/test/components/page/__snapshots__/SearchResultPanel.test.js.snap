// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultPanel testing with snapshot 1`] = `
<div
  className="searchResultPanel"
>
  <div
    className="quantityMovies"
  >
    0 movies found
  </div>
  <button
    className="ratingButtonNoneSelected"
    onClick={[Function]}
  >
    raiting
  </button>
  <button
    className="reliaseDateButtonSelected"
    onClick={[Function]}
  >
    reliase date
  </button>
  <div
    className="sortBy"
  >
    Sort by
  </div>
</div>
`;
